vignettes/relase-process.qmd

```diff
@@ -0,0 +1,198 @@
+  a bump to 0.13.3, but it could also have resulted in 0.14.0 or 1.0.0.  It has
+  not been extensively used in the past, but it is quite useful for tracking
+  incremental changes with different bugfixes and features that may appear. When
+  this development version indicator exists, the documentation site will have an
+  extra `dev/` directory that contains the upcoming changes so that we can
+  continue to develop the hubverse without disrupting the regular documentation
+  flow.
+
+
+## Release Process
+
+When a package is ready for release we use [the checklist](#checklist) at the 
+top of this document. The rationale for this process is described in
+[The proposal discussion for R package release guidelines](https://github.com/orgs/hubverse-org/discussions/29).
+
+### Non-urgent releases only
+
+This release process assumes that we have accumulated bugfixes and/or features
+in the `main` branch, which we are ready to release. If you have a bug that
+needs to be patched _immediately_ and you have new features in the main branch
+that are not yet released, then you should [create a hotfix](hotfixes) instead.
+
+### Releases
+
+Releases are a concept that is specific for GitHub. Under the hood, releases are
+created from commits or tags. When you use the GitHub web interface, you can
+choose to have a tag automatically created from a release. 
+
+Once the release is created on GitHub, then the package will be available on
+the R-Universe in about an hour or less. 
+
+It is good practice to create a tag to release from so that we have a way to
+track the releases even if we eventually move away from GitHub. Even better
+practice is to create _signed_ tags, which adds a cryptographic signature to
+the tag that allows anyone to verify that it came from your computer and not
+someone pretending to be you.
+
+#### Signed Tags
+
+Zhian likes to create tags via the command line because he has set up his git
+configuration to use [a gpg signature](https://www.garrickadenbuie.com/blog/signed-verified-git-commits-keybase-rstudio/#fnref3)
+so the tags and the releases are both verified. Recently, Git and GitHub added
+support for [creating signatures via an SSH key](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification), which is _a lot_ more approchable than GPG.
+
+Tagging and Release can be achieved in a single step with the [github
+cli](https://cli.github.com/) with the command `gh release create X.Y.Z` for
+the version number
+
+```bash
+gh release create 3.3.3
+# ? Title (optional) sandpaper 3.3.3
+# ? Release notes  [Use arrows to move, type to filter]
+#   Write my own
+# > Write using generated notes as template
+#   Leave blank
+```
+
+Selecting "Write using generated notes as a template" opens an editor and
+populates it with the pull requests that have been accepted since the last
+release.
```

Comment:
I think many folks (including myself) use GitHub web interface for releases. Mine are signed because I have listed my GPG key on GitHub so might be nice to provide instructions for that too (as it would mean other online GitHub activities would also be signed). This could be a shorter separate article though.

Reply:
Agreed to split this out to a shorter article.

I also realized that I misspoke. You are correct that your commits are signed on the GitHub interface, but your tags will not be. The releases themselves on GitHub are not signed and you need to explicitly create a signed tag on the CLI. Tags created via the GitHub web interface become [lightweight tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_lightweight_tags), which can easily be moved around (this is how a lot of GitHub actions specify their versions: they move the `v2` tag to new commits as they update).

I created an example of this in https://github.com/zkamvar/improved-enigma/releases and you can see whether or not a tag is signed if it has a green check mark next to it. 

### Release via GitHub interface: unsigned tag

![release showing an unsigned tag on a signed commit](https://github.com/user-attachments/assets/5976c3b0-55d3-4f8c-95ff-320800c62137)

### Release via GitHub interface _after_ creating a signed tag:

I also created a release _after_ created a signed tag:

![release showing a signed tag on a signed commit](https://github.com/user-attachments/assets/7ba37a61-2740-46a7-974c-cb9e7d6690aa)

### Moving a lightweight tag to a new unsigned commit after GitHub Release

And I created a release where I moved a lightweight tag from one signed commit to another unsigned commit:

![a release with a lightweight tag on an unsigned commit with details showing that it was force-pushed](https://github.com/user-attachments/assets/6a808c58-2e2f-47cb-b487-76771991b1d5)

Reply:
I have prioritized mentioning the web interface for releases